pramen-py/src/pramen_py/metastore/reader.py

```diff
@@ -57,8 +58,96 @@ def _apply_uppercase_to_columns_names(
         else:
             return df
 
+    def _extract_dates_from_file_name(
+        self, path: str, target_partition_name: str
+    ) -> Optional[str]:
```

Comment:
The function name says you are extracting date from a file name. If this is the case, the function should return `Optional[datetime.date]`, not `Optional[str]`.

But if you want to return a string, you can rename the function to `_extract_partition_value_from_file_name()`